src/capi/azext_capi/custom.py

```diff
@@ -630,6 +610,85 @@ def create_workload_cluster(  # pylint: disable=too-many-arguments,too-many-loca
     return show_workload_cluster(cmd, capi_name)
 
 
+def install_cni(cmd, cluster_name, workload_cfg, windows, args, user_provided_template=None):
+    # Find the provisioned CIDR range(s) and pass to the CNI chart.
+    cidr0, cidr1 = "", ""
+    attempts, delay = 10, 3
+    with Spinner(cmd, "Finding provisioned CIDRs", "✓ Found provisioned CIDRs"):
+        command = ["kubectl", "get", "cluster", cluster_name, "-o=jsonpath={.spec.clusterNetwork.pods.cidrBlocks[0]}"]
+        for _ in range(attempts):
```

Comment:
instead of having to add the for loop for every different command we run, would it be possible to add the retries withing a run_shell_command_with_retries function that calls run_shell_command in a for loop?

Reply:
Excellent idea–I did that in the latest commit and I'm testing it now.